content/blog/integrating-k8sgpt-to-empower-kubernetes-with-ai-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,427 @@
+
+
+
+### Set up LocalAI backend
+
+
+
+This section will focus on setting up and utilizing LocalAI with a supported LLM model in the local environment. This LocalAI setup in the workstation will be integrated with K8sGPT.
+
+
+
+
+#### Download a LLM model
+
+The LocalAI supports a list of LLM models, such as *LLama*, *GPT4ALL*, *Alpaca* and *koala*, etc. In this blog post, the LLM model *`Llama-2–13b-chat-hf`* from [Hugging Face]( https://huggingface.co/) will be downloaded and used as the local AI backend for K8sGPT.
+
+
+
+After you create an account in *Hugging Face*, log in to the site and share your contact information. You will then be granted access to this model. 
+
+![](/img/hf-llama.png)
+
+Type the following command to clone the LLM model. Make sure you have [*git-lfs*](https://git-lfs.github.com) installed. 
+
+
+
+```shell
+$ git-lfs clone https://huggingface.co/meta-llama/Llama-2-13b-chat-hf
+
+$ tree Llama-2-13b-chat-hf/
+Llama-2-13b-chat-hf/
+├── config.json
+├── generation_config.json
+├── LICENSE.txt
+├── model-00001-of-00003.safetensors
+├── model-00002-of-00003.safetensors
+├── model-00003-of-00003.safetensors
+├── model.safetensors.index.json
+├── pytorch_model-00001-of-00003.bin
+├── pytorch_model-00002-of-00003.bin
+├── pytorch_model-00003-of-00003.bin
+├── pytorch_model.bin.index.json
+├── README.md
+├── Responsible-Use-Guide.pdf
+├── special_tokens_map.json
+├── tokenizer_config.json
+├── tokenizer.json
+├── tokenizer.model
+└── USE_POLICY.md
+
+0 directories, 18 files
+```
+
+#### Install *Text Generation Web UI*
+
+There are many LLM serving framework and tools that support various LLM models and provide a way to interact with those models with OpenAI compatible API endpoints. 
+
+
+
+This section introduces the [Text Generation Web UI (TGW)]( https://github.com/oobabooga/text-generation-webui) tool and shows how to set up this tool to support the locally downloaded LLM model as the AI backend. The TGW is a [Gradio]( https://github.com/gradio-app/gradio/) based tool that builds a web UI and supports many large language model formats with OpenAI compatible APIs.
+
+
+
+Type the following commands to clone the TGW repo and install it in the workstation:
+
+
+
+```shell
+git clone https://github.com/oobabooga/text-generation-webui
+cd text-generation-webui
+pip install -r requirements.txt
+```
+#### Serve local LLM model
+
+
+
+
+Type the following command to start serving the downloaded LLM model in your workstation. The option *'--extensions openai'* specifies to use the OpenAI extension to provide OpenAI format API. The options *'--cpu'* & *'--load-in-4bit'* to load model in 4-bit precision and use the model performed on a CPU to make predictions. This is more cost-effective for inference, and especially it’s helpful in case you don’t have GPU installed in your environment.

```

Comment:
```suggestion
Type the following command to start serving the downloaded LLM model in your workstation. The option *'--extensions openai'* specifies to use the OpenAI extension to provide OpenAI format API. The options *'--cpu'* & *'--load-in-4bit'* to load model in 4-bit precision and use the model performed on a CPU to make predictions. This is more cost-effective for inference, and it’s helpful in case you don’t have GPU's installed in your environment.

```